lib/elixir_analyzer/exercise_test/assert_call/compiler.ex

```diff
@@ -94,12 +104,27 @@ defmodule ElixirAnalyzer.ExerciseTest.AssertCall.Compiler do
     {node, acc} = find(node, acc)
 
     if function_def?(node) do
-      {node, %{acc | in_function_def: nil}}
+      {node, %{acc | in_function_def: nil, in_function_modules: %{}}}
     else
       {node, acc}
     end
   end
 
+  @doc """
+  While traversing the AST, check nodes for uses of `import` or `alias` to keep track of modules in scope
+  """
+  @spec add_directive(Macro.t()) :: nil | [{[atom], [atom]}]
+  def add_directive({:import, _, [module_path | _]}),
+    do: get_paths(module_path, :import)
+
+  def add_directive({:alias, _, [module_path]}),
+    do: get_paths(module_path, :alias)
+
+  def add_directive({:alias, _, [module_path, [as: {:__aliases__, _, [alias]}]]}),
+    do: get_paths(module_path, :alias) |> Enum.map(fn {_, path} -> {[alias], path} end)
+
+  def add_directive(_), do: []
+
```

Comment:
I like the idea of this function bu I have a couple suggestions:
1) this function is doing two things, which might be why the name of the function `add_directive` doesn't seem to quite fit. It is detecting and tracking imports and detecting and tracking aliases, I think, while similar, it is worthwhile to handle them separately
   - Suggestion: split these up even if it isn't as DRY if it makes more sense

     ```elixir
     # I think it would flow well if it could be used in the annotate function like this to replace lines 79-85:
     acc =
       node
       |> trackImports(node, acc)
       |> trackAliases(node, acc)
     ```

2) it seems to depend on the reducer in the `annotate/2` function on line 81-85, so that should probably be inlined into this function as a private helper function

Reply:
```
trackImports
```
Hahaha, have you been coding in... mmh... JavaScript maybe?

Reply:
My daily work is all JavaScript and PHP, so yep